libs/platform/experience/src/plugins/theme/theme.ts

```diff
@@ -43,14 +50,20 @@ export class ThemePlugin extends ThemeTokens implements AppPlugin {
     return this.icons;
   }
 
-  getBreakpoints(): ThemeBreakpoints {
-    return this.breakpoints as ThemeBreakpoints;
+  getBreakpoints(): Breakpoints {
+    return this.breakpoints as Breakpoints;
   }
 
   getIcon(icon: string): string | Promise<string> {
     return resolveLazyLoadable(this.icons[icon]);
   }
 
+  beforeApply(app: App): void {
+    const components = app.findPlugin(ComponentsPlugin);
+    components?.setResolver({ themes: this.resolve.bind(this) });
+    components?.extendComponent(this.applyThemes.bind(this));
```

Comment:
Use `requirePlugin` to enforce `ComponentsPlugin` and remove optional chaining as ThemePlugin cannot working without it anyway:
```ts
    const components = app.requirePlugin(ComponentsPlugin);
    components.setResolver({ themes: this.resolve.bind(this) });
    components.extendComponent(this.applyThemes.bind(this));
```